core/src/utils/native/keyboard.ts

```diff
@@ -18,7 +20,9 @@ export const Keyboard = {
   },
   getResizeMode(): Promise<KeyboardResizeOptions | undefined> {
     const engine = this.getEngine();
-    if (!engine || !engine.getResizeMode) {
+    if (!isPlatform('ios') || !engine?.getResizeMode) {
```

Comment:
Are we able to check the `UNIMPLEMENTED` value from Capacitor instead of checking the platform?

So something like this:

```js
try {
  return engine.getResizeMode();
} catch(e) {
  if (e.code === 'UNIMPLEMENTED') {
    return Promise.resolve(undefined); 
  }
  
  // throw the error unchanged if not an unimplemented error
  throw e;
}
```

My concern with the proposed approach is we are coupling our code to implementation details of the Capacitor plugin. Additionally, `isPlatform` relies on user agent sniffing which can be spoofed, so that check may not always be correct. Checking the unimplemented state lets us rely on Capacitor to tell us about functionality.

Reply:
Ooh that's brilliant, I didn't think of that approach!

Yes, we can do something very similar to this. I ended up using a `.catch` on the original promise, so when it is chained in the implementation, it will resolve `undefined`. It'll swallow the error in the client runtime, but still indicate the unimplemented method invocation in the native bridge logs. 

This is a slightly different net result that the original implementation, which doesn't log the `console.error` for the unimplemented method invocation (because it isn't executed), but solves the original problem of the uncaught error. 

![CleanShot 2023-05-23 at 23 14 12@2x](https://github.com/ionic-team/ionic-framework/assets/13732623/8b0b379e-de86-42fd-91e3-b4ce563a46f2)